main/calculations.py

```diff
@@ -0,0 +1,147 @@
+
+    def status(self, proxy=False):
+        try:
+            if proxy:
+                r = ProxyRequests(self.url)
+            else:
+                r = requests.get(self.url, headers={
+                    'User-Agent': 'Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1)'})
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_success(proxy, r)
+            else:
+                return self.create_new_url_success(proxy, r)
+        except Exception as e:
+            self.error = str(e)
+            if not proxy and config.PROXY:
+                return self.status(proxy=True)
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_exception(e)
+            else:
+                return self.create_url_exception()
+
+    def create_url_exception(self):
+        data = dict()
+        SiteToCheck.objects.create(url=self.url,
+                                   user=self.user,
+                                   last_status=None,
+                                   last_response_time=None,
+                                   last_check=datetime.now().strftime("%Y-%m-%d %H:%M"),
+                                   bad_data=str(
+                                       datetime.now().strftime("%Y-%m-%d %H:%M")) + ': The url is not responding'
+                                   )
+        data['last_status'] = None
+        data['last_response_time'] = None
+        data['bad_data'] = 'The url is not responding'
+        return data
+
+    def modify_url_exception(self, e):
+        data = dict()
+        obj = SiteToCheck.objects.get(url=self.url, user=self.user)
+        obj.last_status = None
+        obj.last_response_time = None
+        obj.last_check = datetime.now().strftime("%Y-%m-%d %H:%M")
+        if len(obj.bad_data) > 0:
+            obj.bad_data += '\n'
+        obj.bad_data += str(datetime.now().strftime("%Y-%m-%d %H:%M")) + ': ' + self.error
+        obj.save()
+        data['bad_data'] = e
+        data['last_status'] = None
+        data['last_response_time'] = None
+        data['last_check'] = datetime.now().strftime("%Y-%m-%d %H:%M")
+        data['url'] = self.url
+        return data
+
+    def create_new_url_success(self, proxy, r):
+        data = dict()
+        data['last_status'] = r.status_code if not proxy else r.get_status_code()
+        data['last_response_time'] = r.elapsed.total_seconds() if not proxy else self.time
+        data['last_check'] = datetime.now().strftime("%Y-%m-%d %H:%M")
+        SiteToCheck.objects.create(url=self.url,
+                                   user=self.user,
+                                   last_status=data['last_status'],
+                                   last_response_time=data['last_response_time'],
+                                   last_check=data['last_check'])
+
+        return data
+
+    def modify_url_success(self, proxy, r):
```

Comment:
To też mogło by być metodą na obiekcie `SiteToCheck`

Reply:
Zrobione - mam nadzieję, że o to chodzi.